golden/config.go

```diff
@@ -174,20 +178,31 @@ func (config Config) entrypoint(inputPath string) (*exec.Cmd, string, error) {
 	isCustom := config.ExecutionConfig != nil
 	args := config.Args
 
+	// Adapt input path, if using custom working directory
+	if isCustom && config.ExecutionConfig.WorkDir != "" {
+		cwd, err := os.Getwd()
+		if err != nil {
+			return nil, "", err
+		}
+		inputPath = filepath.Join(cwd, inputPath)
+	}
+	if _, err := os.Stat(inputPath); os.IsNotExist(err) {
+		return nil, "", fmt.Errorf("input file does not exist: %s", inputPath)
+	}
+
+	// When not handling I/O with stdin/stdout, use temporary files
 	if !config.UseStdIn {
 		inputFlag := "-runner.input.path"
 		if isCustom {
 			inputFlag = config.ExecutionConfig.InputFlag
 		}
 		args = append(args, inputFlag, inputPath)
 	}
-
```

Comment:
Matter of taste, I like a newline after a control block finishes (for, if, etc)

Reply:
The idea here was to group the different steps nicely. But yeah, maybe a bit aggressive.